language/testing-infra/e2e-tests/src/executor.rs

```diff
@@ -38,13 +40,17 @@ use move_vm_types::{
 };
 use vm::CompiledModule;
 
+static RNG_SEED: [u8; 32] = [9u8; 32];
+
 /// Provides an environment to run a VM instance.
 ///
 /// This struct is a mock in-memory implementation of the Libra executor.
 #[derive(Debug)]
 pub struct FakeExecutor {
     data_store: FakeDataStore,
     block_time: u64,
+    executed_output: Option<GoldenOutputs>,
+    rng: KeyGen,
```

Comment:
It feels a little bit awkward to me to add these to the FakeExecutor. Would it be better if we move this logic to the outside by implementing a wrapper or a few helper functions? Hardcoding this into the FakeExecutor will make it harder for other tests/libraries to depend on it.

Reply:
The main reason is to reduce the changes needed for each individual tests. You could imagine that otherwise we will need to create a seed for each tests. I also tried to have one global seed, but it didn't work either as cargo will run the tests together concurrently.

Reply:
> The main reason is to reduce the changes needed for each individual tests.
Understood and agreed. 

> You could imagine that otherwise we will need to create a seed for each tests.
No doubt about this.

Read the code again and I'm starting to understand what's causing our "disagreement" here. I feel like we were seeing `FakeExecutor` in two different ways:
- Me: it's a mock LibraExecutor/LibraVM, so it should do nothing other than executing Libra transactions
- You: it's modeling a testing flow so it should own the RNG and do other stuff like injecting accounts into the storage

I think your understanding is right considering the actual ownership model and what the methods are doing, so I'd be happy to stamp this diff. However I do realize this exposes a few other problems and we should try to get them solved:
- Rename `FakeExecutor` to something like `xxxTestingFlow` or `xxxTestingEnv`. With `Executor` being part of its name, it made me think this is more like an instance implementing the `VMExecutor` trait, mocking the real `LibraExecutor`, which is not true.
- For the immediate future, functional tests should not depend on this as it models a different testing flow. Instead it should use the fake storage and the LibraVM APIs directly. (Note: don't bother fixing the existing implementation. There's a rewriting coming in a month or two, in a style similar to #6540)